jsonnet/docker/plugin.libsonnet

```diff
@@ -0,0 +1,117 @@
+local windows_pipe = '\\\\\\\\.\\\\pipe\\\\docker_engine';
+local windows_pipe_volume = 'docker_pipe';
+
+local windows_settings = {
+  daemon_off: true,
+  // Workaround for https://github.com/drone/drone-cli/issues/117
+  purge: 'false',
+};
+
+{
+  /**
+   * Returns a Docker plugin using the given settings.
+   *
+   * @param name The name of the pipeline step.
+   * @param settings The settings for the plugin.
+   * @return A json representing a Docker plugin and its settings.
+   */
+  Plugin(name, settings): {
```

Comment:
Declaring a plugin should be consistent. My thought is that there's always a `Plugin` function that'll create the plugin.

The thing I'm not sure of here is that there's always a `settings` variable since the settings are nested within the plugin's definition.

Plugins always look like this.

```
- name: foo
   image: plugins/foo
   settings:
     foo: false
     bar: true
```

Also maybe it just always sets the defaults? Or always has the required parameters in it.

Reply:
It's really personal taste, but I like using function params for the "identity" of the thing:

```
Plugin(name) {
  settings+: {
    ...
  },
}
```